src/AddressSpace.cc

```diff
@@ -1817,23 +1817,25 @@ bool AddressSpace::has_exec_watchpoint_fired(remote_code_ptr addr) {
 }
 
 bool AddressSpace::allocate_watchpoints() {
```

Comment:
Have you given any thought to how this will work on AArch64? I would think you could shoehorn most of the arch specific stuff into Task::set_debug_regs and this code could be largely unchanged.

Reply:
No, I haven't looked into watchpoints yet.

Reply:
If you'd prefer I can drop this part of it for now until we figure out how to do watchpoints properly.

Reply:
I think that would be good.